libctru/source/services/irrst.c

```diff
@@ -39,6 +40,18 @@ Result irrstInit(u32* sharedMem)
 
 	// Map ir:rst shared memory at addr "sharedMem".
 	irrstSharedMem=sharedMem;
+	if(!irrstSharedMem)
```

Comment:
irrstInit's sharedMem parameter should have been removed, instead of preserving it and managing ownership.